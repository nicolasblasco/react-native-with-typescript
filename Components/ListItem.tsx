import React from 'react';
import { View, Text, StyleSheet } from 'react-native';

interface Props {
  id: number;
  name: string;
  email: string;
}

const ListItem: React.FC<Props> = ({id, name, email}) => {
    return (
      <View style={styles.item}>
        <Text style={styles.data}>ID: {id}</Text>
        <Text style={styles.data}>Name: {name}</Text>
        <Text style={styles.data}>Email: {email}</Text>
      </View>
    )
};

const styles = StyleSheet.create({
  item: {
    margin: 5,
    padding: 10,
    backgroundColor: '#11698E'
  },
  data: {
    fontSize: 15,
    color: '#FFF'
  }
});

export default ListItem;