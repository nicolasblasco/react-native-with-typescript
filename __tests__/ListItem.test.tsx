import 'react-native';
import React from 'react';
import ListItem from '../src/Components/Shared/Item';

import {render, fireEvent} from '@testing-library/react-native';

const id = 1;
const email = 'test@gmail.com';
const name = 'test';

describe('ListItem', () => {
  it('Renders correctly', async () => {
    const onDelete = jest.fn();
    const onUpdate = jest.fn();

    const {getByText, getByTestId, toJSON} = render(
      <ListItem
        id={id}
        email={email}
        name={name}
        onDelete={onDelete}
        onUpdate={onUpdate}
      />,
    );

    const updateButton = getByTestId('update-button');
    fireEvent.press(updateButton);

    const deleteButton = getByTestId('delete-button');
    fireEvent.press(deleteButton);

    expect(getByText(`Name: ${name}`)).toBeTruthy();
    expect(onDelete).toBeCalled();
    expect(onUpdate).toBeCalled();
    expect(toJSON()).toMatchSnapshot();
  });
});
