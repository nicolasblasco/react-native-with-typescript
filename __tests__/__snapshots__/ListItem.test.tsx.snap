// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ListItem Renders correctly 1`] = `
<View
  style={
    Object {
      "backgroundColor": "#F8F1F1",
      "borderLeftColor": "#19456B",
      "borderLeftWidth": 5,
      "flexDirection": "row",
      "justifyContent": "space-between",
      "margin": 5,
      "padding": 10,
    }
  }
>
  <View>
    <Text
      style={
        Object {
          "color": "#19456B",
          "fontSize": 15,
        }
      }
    >
      ID: 
      1
    </Text>
    <Text
      style={
        Object {
          "color": "#19456B",
          "fontSize": 15,
        }
      }
    >
      Name: 
      test
    </Text>
    <Text
      style={
        Object {
          "color": "#19456B",
          "fontSize": 15,
        }
      }
    >
      Email: 
      test@gmail.com
    </Text>
  </View>
  <View
    style={
      Object {
        "justifyContent": "space-between",
      }
    }
  >
    <View
      accessible={true}
      collapsable={false}
      focusable={true}
      nativeID="animatedComponent"
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "alignItems": "center",
          "backgroundColor": "#FFFFFF",
          "borderColor": "#11698E",
          "borderRadius": 3,
          "borderWidth": 1,
          "opacity": 1,
          "padding": 3,
        }
      }
      testID="update-button"
    >
      <Text
        style={
          Object {
            "color": "#11698E",
          }
        }
      >
        Update
      </Text>
    </View>
    <View
      accessible={true}
      collapsable={false}
      focusable={true}
      nativeID="animatedComponent"
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "alignItems": "center",
          "backgroundColor": "#DA1212",
          "borderRadius": 3,
          "opacity": 1,
          "padding": 3,
        }
      }
      testID="delete-button"
    >
      <Text
        style={
          Object {
            "color": "#FFFFFF",
          }
        }
      >
        Delete
      </Text>
    </View>
  </View>
</View>
`;
